LibAddonMenu-2.0/LibAddonMenu-2.0/controls/editbox.lua

```diff
@@ -15,12 +17,28 @@
 } ]]
 
 
-local widgetVersion = 14
+local widgetVersion = 15
 local LAM = LibStub("LibAddonMenu-2.0")
 if not LAM:RegisterWidget("editbox", widgetVersion) then return end
 
 local wm = WINDOW_MANAGER
 
+local function GetValidTextType(textType)
+    textType = LAM.util.GetDefaultValue(textType)
+    if textType and (textType >= TEXT_TYPE_ITERATION_BEGIN and textType <= TEXT_TYPE_ITERATION_END) then

```

Comment:
Wouldn't it make more sense to check for the type: `type(textType) == "number"`, otherwise it could throw an error when the textType is compared. It may also be worth to invert the statement (if type ~= number or type < begin or type > end then return type_all).

Reply:
i've changed to for number check. but if you do ~= number and then check for number, it will throw an error

Reply:
I meant like this:
```
local function GetValidTextType(textType)
    textType = LAM.util.GetDefaultValue(textType)
    if type(textType) ~= "number" or textType < TEXT_TYPE_ITERATION_BEGIN or textType > TEXT_TYPE_ITERATION_END then
        return TEXT_TYPE_ALL
    end
    return textType
end
```
That way the if condition can return early in case one of the conditions doesn't evaluate to true.